tests/jax/components/building/networks_test.py

```diff
@@ -0,0 +1,209 @@
+    test_builder.store.agent_net_keys = {
+        "net_key1": "network_1",
+        "net_key2": "network_2",
+    }
+    return test_builder
+
+
+@pytest.fixture
+def test_default_networks(test_network_factory: Callable) -> Networks:
+    """Pytest fixture for default networks.
+
+    Args:
+        test_network_factory: factory to use in default networks config.
+
+    Returns:
+        Default networks test component.
+    """
+    networks_config = NetworksConfig()
+    networks_config.network_factory = test_network_factory
+    networks_config.seed = 919
+
+    return DefaultNetworks(networks_config)
+
+
+def test_key_in_store(
+    test_default_networks: Networks, test_builder: SystemBuilder
+) -> None:
+    """Test if key is loaded into the store by build.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    assert not hasattr(test_builder.store, "key")
+    test_default_networks.on_building_init_start(test_builder)
+    assert hasattr(test_builder.store, "key")
+    assert isinstance(test_builder.store.key, jax.random.PRNGKeyArray) or isinstance(
+        test_builder.store.key, jax.numpy.DeviceArray
+    )
+
+
+def test_config_set(test_default_networks: Networks) -> None:
+    """Test if config is set by component init.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+
+    Returns:
+        None.
+    """
+    assert test_default_networks.config.seed == 919
+
+
+def test_network_factory_environment_spec(
+    test_default_networks: Networks, test_builder: SystemBuilder
+) -> None:
+    """Test if environment spec is given to the network factory and stored by build.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    test_default_networks.on_building_init_start(test_builder)
+    networks = test_builder.store.network_factory()
+
+    for network in networks.values():
+        assert network["environment_spec"]._keys == ["agent_0", "agent_1"]
+        assert network["environment_spec"]._specs["agent_0"].observations.shape == (
+            10,
+            5,
+        )
+
+
+def test_network_factory_agent_net_keys(
+    test_default_networks: Networks, test_builder: SystemBuilder
+) -> None:
+    """Test if net keys are given to the network factory and stored by build.
+
+    Args:
+        test_default_networks: Pytest fixture for default networks component.
+        test_builder: Pytest fixture for test system builder
+
+    Returns:
+        None.
+    """
+    test_default_networks.on_building_init_start(test_builder)
+    networks = test_builder.store.network_factory()
+
+    for network in networks.values():
+        assert network["agent_net_keys"] == {
+            "net_key1": "network_1",
```

Comment:
Same as above.

Reply:
I defined these myself in this file, so it does not apply here. See function `test_builder()` on line 73/74.